compose/parallel.py

```diff
@@ -314,10 +315,10 @@ def _write_noansi(self, msg, obj_index, status):
     def write(self, msg, obj_index, status, color_func):
         if msg is None:
             return
-        if self.noansi:
-            self._write_noansi(msg, obj_index, status)
-        else:
+        if self.ansi == AnsiMode.always or (self.ansi == AnsiMode.auto and self.stream.isatty()):
```

Comment:
```suggestion
        if self.ansi.use_tty(self.stream):
```
Or other method name defined in the enum